tools/run_tests/xds_k8s_test_driver/framework/infrastructure/traffic_director.py

```diff
@@ -290,6 +292,28 @@ def delete_forwarding_rule(self, force=False):
         self.compute.delete_forwarding_rule(name)
         self.forwarding_rule = None
 
+    def create_firewall_rule(self, allowed_ports: List[str]):
+        name = self._ns_name(self.FIREWALL_RULE_NAME)
+        logging.info(
+            'Creating firewall rule "%s" in network "%s" with allowed ports %s',
+            name, self.network, allowed_ports)
+        resource = self.compute.create_firewall_rule(
+            name, self.network_url, xds_flags.FIREWALL_SOURCE_RANGE.value,
+            allowed_ports)
+        self.firewall_rule = resource
```

Comment:
![image](https://user-images.githubusercontent.com/672669/118729522-13699a80-b804-11eb-8fd8-ef566c93e9d9.png)

To make the linter happy, let's declare `self.firewall_rule` (and define the type) in `__init__` (after `self.url_map`) :

```
self.firewall_rule: Optional[GcpResource] = None
```

Reply:
Done. Good catch!